### ISSUE TITLE
In-Game: lengthdir_y() returns values with the wrong sign when using literal numbers in 2024.800 runtimes

### ISSUE THREAD

**Issue by gnysek** [YoYoGames/GameMaker-Bugs#7449]
### Description
In 2024.800 betas it seems that `lengthdir_y` returns opposite data (like `* -1` would be applied on it).

For example, `lengthdir_y(5, 270)` returns -5, but 270 directs to South, so it should be +5.

### Steps To Reproduce
Run:
```gml
show_debug_message(lengthdir_y(5, 0));
show_debug_message(lengthdir_y(5, 45)); // should be -3.54, but returns 3.45
show_debug_message(lengthdir_y(5, 90)); // should be -5, but returns 5
show_debug_message(lengthdir_y(5, 270)); // should be 5, but returns -5
```
and compare results

### Which version of GameMaker are you reporting this issue for?
IDE v2024.800.0.609 Runtime v2024.800.0.633

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

### Which platform(s) are you seeing the problem on?
Windows

[0dcd34d9-dc5e-4f9d-84eb-dc4dc4b7ad4e](https://api.gamemaker.io/api/github/downloads/0dcd34d9-dc5e-4f9d-84eb-dc4dc4b7ad4e)

Edit: it happens with constants only, so it must be some recent GM optimization.
When doing:
```gml
show_debug_message(lengthdir_y(5, 270)); // should be 5
var a = 5;
show_debug_message(lengthdir_y(a, 270)); // should be 5
```
it returns:
```
-5
5
```

**Comment by KormexGit** [comment]
The optimization for math functions with number literals was added in 2024.6, and lengthdir_y returns the correct values in monthly, so it somehow broke during beta

**Comment by rwkay** [comment]
Fixed in 2024.8

**Comment by YYDan** [comment]
Note that both _x and _y were also changed to truncate, rather than round, when calculating their results.

**Comment by scott-dunbar** [comment]
Verified fixed in  IDE v2024.800.0.618  Runtime v2024.800.0.642
